packages/react-charting/src/utilities/utilities.ts

```diff
@@ -313,26 +415,71 @@ export function createYAxis(yAxisParams: IYAxisParams, isRtl: boolean, axisData:
   const finalYmin = yMinMaxValues.startValue < yMinValue ? 0 : yMinValue!;
   const domainValues = prepareDatapoints(finalYmax, finalYmin, yAxisTickCount);
   const yAxisScale = d3ScaleLinear()
-    .domain([finalYmin, domainValues[domainValues.length - 1]])
+    .domain([finalYmin, finalYmax])
     .range([containerHeight - margins.bottom!, margins.top! + (eventAnnotationProps! ? eventLabelHeight! : 0)]);
   const axis = isRtl ? d3AxisRight(yAxisScale) : d3AxisLeft(yAxisScale);
   const yAxis = axis
     .tickPadding(tickPadding)
     .tickValues(domainValues)
     .tickSizeInner(-(containerWidth - margins.left! - margins.right!));
   yAxisTickFormat ? yAxis.tickFormat(yAxisTickFormat) : yAxis.tickFormat(d3Format('.2~s'));
+
   yAxisElement ? d3Select(yAxisElement).call(yAxis).selectAll('text').attr('aria-hidden', 'true') : '';
   axisData.yAxisDomainValues = domainValues;
   return yAxisScale;
 }
 
+export const createStringYAxis = (
```

Comment:
how can we avoid so much code duplication

Reply:
The problem in reducing the code duplication here is that almost all the different functions having different arguments, We  need to spend some time on the  utilities code to remove code duplication. I will create a work item for this.